### PR TITLE
Legion Rework

### PR DESCRIPTION
**Pull request by Edan52**
## What Does This PR Do
Tweaks the legion megafauna quite a bit, making the fight more difficult and more interesting.

Here is what has changed:

- Legion now uses the animated sprite
- It has the same charge as before
- It no longer splits
- The legion now has 2500 health
- It now does the same damage as other megafauna
- The sprite gets smaller as the fight goes on
- It summons mini legion skulls each time it gets hit

I have tested this locally and this is killable with both crusher and KA without cheese.

Charge:
Charge is now a bit faster, and is more deadly due to the damage change, if you don't react fast you'll most likely get hit.

Death laser:
One of the new attacks is a tile wide laser which telegraphs where it will strike before firing, it gives between 1.25 and 0.25 seconds of time before firing based on its health, giving less time to react as it gets lower on health. The laser deals 50-70 burn damage, dealing more damage the lower health the Legion is, this laser fully respects laser armour and has no penetration. The laser deals damage upon firing and will deal damage again when walked into, I may lower the damage it deals if you walk into it. In most cases, you won't make the mistake of walking back into the laser and the initial laser can generally be dodged just by moving diagonally. The laser will dust dead mobs which aren't part of the lavaland faction. It seems scary but is generally not a major threat in the fight, but it can catch you off guard if you aren't moving correctly or are distracted or towards the end of the fight it fires with a tiny warning, this can be very hard to dodge if you aren't moving correctly.

Big legion:
I have reworked the big legion and implemented it. With a low percentage change, the legion will create a big legion as an attack, this big legion moves very quickly and after attacking you twice, its third attack will be a ground pound that deals more damage and sends you flying a few tiles.

The Legion also summons regular legion mobs with a higher chance than the big legion, they allow the player to acquire legion cores mid fight and generally the player will never be slowed if they play it right.

Crusher trophy:
The legion now drops a crusher trophy, that will allow you to dash a few tiles after a mark detonation. One cool thing that can be done with this is to dash into pushable mobs to make them face away from you, to get a backstab. Credit to FoS for the sprite.

## Why It's Good For The Game
The current fight is lacklustre, and needs to be more interesting. Currently, you can just make a long hallway and kite it. It's not hard, it just takes a long time, this PR is intended to make the fight more interesting and increase the difficulty. Legion should be a force to be reckoned with.

## Images of changes
Coming soon

## Changelog
:cl: CornMyCob
tweak: The legion has been reworked, it now fires lasers at the player and creates legions as it breaks apart.
/:cl:

### CODE REVIEW

File: code/datums/status_effects/debuffs.dm [hunk 1]
```diff
@@ -137,6 +137,32 @@
...
+		latest_attacker = attacker
+	if(stacks != stack_threshold)
+		return TRUE
+
+/datum/status_effect/stacking/ground_pound/stacks_consumed_effect()
+	flick("legion-smash", latest_attacker)
+	addtimer(CALLBACK(latest_attacker, /mob/living/simple_animal/hostile/asteroid/big_legion/.proc/throw_mobs), 1 SECONDS)
+
+/datum/status_effect/stacking/ground_pound/on_remove()
+	latest_attacker = null
```
Comment (hal9000PR) [inline code/datums/status_effects/debuffs.dm:165]:
```suggestion
	latest_attacker = null

```

File: code/modules/mob/living/simple_animal/hostile/mining/hivelord.dm [hunk 2]
```diff
@@ -279,42 +279,50 @@
...
+/mob/living/simple_animal/hostile/asteroid/big_legion/AttackingTarget()
+	if(!isliving(target))
+		return ..()
+	var/mob/living/L = target
+	var/datum/status_effect/stacking/ground_pound/G = L.has_status_effect(STATUS_EFFECT_GROUNDPOUND)
+	if(!G)
+		L.apply_status_effect(STATUS_EFFECT_GROUNDPOUND, 1, src)
+		. = ..()
+	else if(G.add_stacks(stacks_added = 1, attacker = src))
+		. = ..()
```
Comment (hal9000PR) [inline code/modules/mob/living/simple_animal/hostile/mining/hivelord.dm:310]:
```suggestion
	if(!G)
		L.apply_status_effect(STATUS_EFFECT_GROUNDPOUND, 1, src)
		return ..()
	if(G.add_stacks(stacks_added = 1, attacker = src))
		return ..()
```

**Review by hal9000PR — APPROVED** [review]

**Review by hal9000PR — CHANGES_REQUESTED** [review]
missclick

File: code/modules/mining/equipment/kinetic_crusher.dm [hunk 3]
```diff
@@ -90,6 +90,14 @@
 	. = ..()
 	if(!wielded)
 		return
+	if(user.has_status_effect(STATUS_EFFECT_DASH) && user.a_intent == INTENT_HELP)
+		if(user.throw_at(target, range = 3, speed = 3, spin = FALSE, diagonals_first = TRUE))
+			playsound(src, 'sound/effects/stealthoff.ogg', 50, 1, 1)
+			user.visible_message("<span class='warning'>[usr] dashes!</span>")
```
Comment (Sirryan2002) [inline code/modules/mining/equipment/kinetic_crusher.dm:96]:
```suggestion
			user.visible_message("<span class='warning'>[user] dashes!</span>")
```

File: code/modules/mob/living/simple_animal/hostile/megafauna/legion.dm [hunk 4]
```diff
@@ -61,81 +66,108 @@ Difficulty: Medium
...
-			elimination = FALSE
-		else if(prob(5))
-			loot = list(/obj/structure/closet/crate/necropolis/tendril)
-			if(!true_spawn)
-				loot = null
-		return ..()
+/mob/living/simple_animal/hostile/megafauna/legion/proc/fire_disintegration_laser(location)
+	playsound(loc, 'sound/weapons/marauder.ogg', 200, TRUE)
+	Beam(location, icon_state = "death_laser", time = 2 SECONDS, maxdistance = INFINITY, beam_type = /obj/effect/ebeam/disintegration)
+	for(var/turf/t in getline(src,location))
```
Comment (Sirryan2002) [inline code/modules/mob/living/simple_animal/hostile/megafauna/legion.dm:135]:
```suggestion
	for(var/turf/t in getline(src, location))
```

**Review by Sirryan2002 — COMMENTED** [review]
I don't know enough about mob/megafauna/effects code to be much help here so here's a few formatting suggestions

**Review by hal9000PR — APPROVED** [review]

**Pull request merged**